be/src/exec/base_scanner.cpp

```diff
@@ -260,6 +279,33 @@ Status BaseScanner::_fill_dest_tuple(Tuple* dest_tuple, MemPool* mem_pool) {
     return Status::OK();
 }
 
+Status BaseScanner::filter_block(vectorized::Block* temp_block, size_t slot_num) {
+    // filter src tuple by preceding filter first
+    auto old_rows = temp_block->rows();
```

Comment:
rethink the logic?

Reply:
it should update "old_rows" in the for loop.